### PR TITLE
support named parameters for replacement

### PR DESCRIPTION
**Pull request by tsloughter**
So this is a huge change... But I tried to keep the api the same. I found neotoma easier to write the parser in and it returns iolists, since that is only done once (and not per-run) I do an `iolist_to_binary` on the results so it is readable, but this does change the api to be returning binary strings instead of lists.

In the case of replacing named parameters I left them as iolists in that case, since it shouldn't be necessary to flatten them.

I also added support for using `-- name: the-name` to name the queries, to match yesql, but the format `-- :the-name` still works as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,10 +42,13 @@ In Erlang you can use this queries like functions:
 > {ok, Queries} = eql:compile("queries/user.sql"). % with path to your queries file
 > {ok, Q1} = eql:get_query(get_all_users, Queries).
 > {ok, Q2} = proplists:get_value(get_user_by_id, Queries).
+> {ok, Q3} = eql:get_query(get_by_id, Queries, [{table, "some_table"}]).
 > Q1.
-%> "SELECT * FROM users;"
+%> <<"SELECT * FROM users;">>
```
Comment (artemeff) [inline README.md:47]: `{:ok, <<"SELECT * FROM users;">>}` :)

**Review by artemeff — COMMENTED** [review]
Do you think that we need to warn about sanitization in named params?

**Comment by tsloughter** [comment]
D'oh, I somehow missed that you had reviewed!

Do you mean to just warn in the readme that named params aren't sanitized?

File: README.md (hunk 1 above)
Comment (tsloughter) [reply README.md:47]: The `ok` is matched out in the line above, but the `Q3` is wrong because it kept the `{ok, ...}` so I'm fixing that :)

**Comment by artemeff** [comment]
> Do you mean to just warn in the readme that named params aren't sanitized?

Yep :)

**Comment by artemeff** [comment]
Thank you! ❤️

**Pull request merged**